dev_tools/utils/nxdl_utils.py

```diff
@@ -46,7 +46,7 @@ def decode_or_not(elem, encoding: str = "utf-8", decode: bool = True):
         try:
             return elem.decode(encoding)
         except UnicodeDecodeError as e:
-            raise ValueError(f"Error decoding bytes") from e
+            raise UnicodeDecodeError(f"Error decoding bytes: {e}")
```

Comment:
Just use `e.add_note` (see https://docs.python.org/3/tutorial/errors.html#enriching-exceptions-with-notes). Also update the docstring

Reply:
Using `e.add_note` now, docstring was also updated